### PR TITLE
Fix validation and example generation for opts

### PR DESCRIPTION
**Pull request by pieterbos**
Fixes https://github.com/nedap/archetype-languageserver/issues/12
Fixes https://github.com/nedap/archetype-languageserver/issues/11

Might fix https://github.com/nedap/archetype-languageserver/issues/13

### CODE REVIEW

**Review by EBrader — CHANGES_REQUESTED** [review]
Functionally, this works perfect Pieter. I only have some suggestions to simplify the code. Furthermore, I believe there is room for improvement in `RunCommandTest` by means of parametrisation, because the current tests for `generate` contain quite some duplicate code. As it is a test suite, I leave it up to you what to do with that 😬

File: src/main/java/com/nedap/openehr/lsp/commands/GenerateExampleCommand.java
Comment (pieterbos) [reply src/main/java/com/nedap/openehr/lsp/commands/GenerateExampleCommand.java:52]: It could be, but the code style in Archie never uses single-statement blocks without `{}`, so i prefer to keep this style

**Review by EBrader — CHANGES_REQUESTED** [review]
Technically, spot on! I only have some comments on documentation/structure for which you may decide for yourself what you do with them. If you looked at and/or processed them, let me know via re-review request 🙃

File: src/test/java/com/nedap/openehr/lsp/RunCommandTest.java
Comment (pieterbos) [reply src/test/java/com/nedap/openehr/lsp/RunCommandTest.java:50]: That is not a good idea. openResource can fail if you assert directly after opening resources, due to missing dependencies. Then if you add the missing dependency by opening the other resource, the server will automatically fix these by pushing new diagnostics, which are then correctly handled in the testclient by removing the old one. Your change would make the test depend on the ordering of the loaded archetypes, and that is not what this test is testing.

I will add the diagnostics to the error message, which is a minor improvement since really this is only to check if the author of the test archetypes did not make a mistake, not if validation or diagnostics work properly.

File: src/test/java/com/nedap/openehr/lsp/RunCommandTest.java
Comment (pieterbos) [reply src/test/java/com/nedap/openehr/lsp/RunCommandTest.java:29]: Both forms are correct English, so I really do not care much.

File: src/test/java/com/nedap/openehr/lsp/RunCommandTest.java
Comment (pieterbos) [reply src/test/java/com/nedap/openehr/lsp/RunCommandTest.java:30]: It's to assert the start of the file, I will add 'should have'

File: src/test/java/com/nedap/openehr/lsp/RunCommandTest.java
Comment (EBrader) [reply src/test/java/com/nedap/openehr/lsp/RunCommandTest.java]: Makes sense. Thanks for the good explanation. 👍🏻

**Review by EBrader — APPROVED** [review]
You got yourself an approval! 🥇

**Pull request merged**
